gdx/src/com/badlogic/gdx/graphics/g2d/GlyphLayout.java

```diff
@@ -529,12 +514,23 @@ public String toString () {
 		 * advance relative to previous glyph position. Last entry is the width of the last glyph. */
 		public FloatArray xAdvances = new FloatArray();
 		public float x, y, width;
-		public final Color color = new Color();
+		public IntArray colors = new IntArray(); // rgba8888
 
+		private void appendRun (GlyphRun run) {
+			glyphs.addAll(run.glyphs);
+			for (int i = xAdvances.isEmpty() ? 0 : 1, n = run.xAdvances.size; i < n; i++) {
+				xAdvances.add(run.xAdvances.get(i));
+			}
+			colors.addAll(run.colors);
+		}
+		
 		public void reset () {
 			glyphs.clear();
 			xAdvances.clear();
+			x = 0;
+			y = 0;
```

Comment:
`x` and `y` don't need to be reset for pooling because they don't affect GC. That is assuming they are set by the code that calls `obtain`.